cmd/minikube/cmd/start.go

```diff
@@ -113,6 +115,8 @@ func runStart(cmd *cobra.Command, args []string) {
 	libMachineClient := libmachine.NewClient(constants.Minipath, constants.MakeMiniPath("certs"))
 	defer libMachineClient.Close()
 
+	setProxy()
```

Comment:
There might be a misunderstanding here. I don't want to change the DockerEnv property, but instead of just assigning _dockerEnv_, I'd make this a function call at this point (which returns the correct "proxified" string array if required).

Also, what comes from docker-machine? cluster.MachineConfig is from us right? So we can easily for example add the proxy to it.